### PR TITLE
ADM-731: [frontend] branch list should bring to front

### PR DESCRIPTION
**Pull request by mrcuriosity-tw**
## Summary

Fix
- https://dorametrics.atlassian.net/browse/ADM-731
- https://dorametrics.atlassian.net/browse/ADM-756

## Before
**Screenshots**

ADM-731
image

ADM-756
image

## After
**Screenshots**
image (1)

## Note

_Null_

### CODE REVIEW

File: frontend/src/containers/MetricsStep/Classification/index.tsx [hunk 1]
```diff
@@ -1,56 +1,111 @@
...
-  flag: boolean;
-}
 export interface classificationProps {
   title: string;
   label: string;
-  targetFields: TargetFieldType[];
+  targetFields: ITargetFieldType[];
 }
 
+export const suffixForDuplicateNames = (targetFields: ITargetFieldType[]) => {
```
Comment (SimonTal) [inline frontend/src/containers/MetricsStep/Classification/index.tsx:17]: Try try this, less loop and simpler
```
export const suffixForDuplicateNames = (targetFields: ITargetFieldType[]) => {
  const nameSumMap = new Map<string, number>();
  const nameCountMap = new Map<string, number>();

  targetFields.forEach((item) => {
    const name = item.name;
    const count = nameCountMap.get(item.name) || 0;
    nameSumMap.set(name, count + 1);
    nameCountMap.set(name, count + 1);
  });

  return targetFields.map((item) => {
    const newItem = { ...item };
    const name = newItem.name;
    const count = nameCountMap.get(newItem.name) || 0;
    const maxCount = nameSumMap.get(newItem.name) || 0;

    if (maxCount > 1) {
      newItem.name = `${newItem.name}-${maxCount - count + 1}`;
      nameCountMap.set(name, count - 1);
    }

    return newItem;
  });
};
```

File: frontend/src/containers/MetricsStep/Classification/index.tsx [hunk 2]
```diff
@@ -1,56 +1,111 @@
...
+    return selectedOptions.length > 0 && selectedOptions.length === targetFieldsWithSuffix.length;
+  }, [selectedOptions, targetFieldsWithSuffix]);
+
+  const handleChange = (_: React.SyntheticEvent, value: ITargetFieldType[]) => {
+    let nextSelectedOptions: ITargetFieldType[];
+    if (value.length === 0) {
+      nextSelectedOptions = [];
+    } else {
+      nextSelectedOptions =
+        value[value.length - 1].key === 'all' ? (isAllSelected ? [] : [...targetFieldsWithSuffix]) : [...value];
```
Comment (SimonTal) [inline frontend/src/containers/MetricsStep/Classification/index.tsx:60]: extract 'all' to a Const ?

File: frontend/src/containers/MetricsStep/Classification/index.tsx [hunk 3]
```diff
@@ -1,56 +1,111 @@
...
+          };
+          return [allOption, ...filtered];
+        }}
+        getOptionLabel={(option) => option.name}
+        onChange={(event, value) => handleChange(event, value as ITargetFieldType[])}
+        renderOption={(props, option: any, state) => {
+          const selectAllProps = option.key === 'all' ? { checked: isAllSelected } : {};
+          return (
+            <li {...props}>
+              <Checkbox style={{ marginRight: 8 }} checked={state.selected} {...selectAllProps} />
```
Comment (SimonTal) [inline frontend/src/containers/MetricsStep/Classification/index.tsx:94]: rem for marginRight?

File: frontend/src/containers/MetricsStep/Classification/index.tsx (hunk 1 above)
Comment (mrcuriosity-tw) [reply frontend/src/containers/MetricsStep/Classification/index.tsx:17]: Cool, elegant!

File: frontend/src/containers/MetricsStep/Classification/index.tsx (hunk 2 above)
Comment (mrcuriosity-tw) [reply frontend/src/containers/MetricsStep/Classification/index.tsx:60]: make sense.

File: frontend/src/containers/MetricsStep/Classification/index.tsx (hunk 3 above)
Comment (mrcuriosity-tw) [reply frontend/src/containers/MetricsStep/Classification/index.tsx:94]: Just copied the legacy code.
Fixed.

**Review by SimonTal — APPROVED** [review]
Approved

**Pull request merged**